src/caches/bdf_caches.jl

```diff
@@ -334,8 +336,10 @@ function alg_cache(alg::QNDF{MO},u,rate_prototype,uEltypeNoUnits,uBottomEltypeNo
   u₀ = zero(u)
   dtprev = one(dt)
   u₀ = zero(u)
-  D = zero(similar(u, length(u), max_order + 2))
-  Dtmp = zero(similar(D))
+  D = similar(u, length(u), max_order + 2)
```

Comment:
Wouldn't it be better to use
```suggestion
  D = similar(u, uEltypeNoUnits, length(u), max_order + 2)
```
or
```suggestion
  D = Matrix{uEltypeNoUnits}(undef, length(u), max_order + 2)
```
as above since `D` is filled with `zero(uEltypeNoUnits)`?

Reply:
Yes, I wasn't being consistent because implicit solvers don't support units anyway.